src/charm.py

```diff
@@ -188,16 +189,20 @@ def _remove_unlisted_plugins(self, container: ops.Container) -> ops.StatusBase:
             return ops.BlockedStatus("Failed to remove plugins.")
         return ops.ActiveStatus()
 
-    def _on_update_status(self, _: ops.UpdateStatusEvent) -> None:
+    def _on_update_status(self, event: ops.UpdateStatusEvent) -> None:
         """Handle update status event.
 
         On Update status:
         1. Remove plugins that are installed but are not allowed by plugins config value.
         2. Update Jenkins patch version if available and is within restart-time-range config value.
+
+        Args:
+            event: The update status event.
         """
         container = self.unit.get_container(JENKINS_SERVICE_NAME)
         if not jenkins.is_storage_ready(container):
             self.unit.status = ops.WaitingStatus("Waiting for container/storage.")
+            event.defer()
```

Comment:
The update status event occurs periodically at regular intervals..Wouldn't it be risky to defer it?

Reply:
Yeh I think in this case we should probably not defer